ale/drivers/mex_drivers.py

```diff
@@ -322,77 +322,162 @@ def detector_center_sample(self):
+        The ephemeris times start at the first bye of the line/records and are 8 bytes.
+
+        Returns
+        -------
         : list
-          Line times
+          Ephemeris times
+        """
+        if not hasattr(self, '_binary_ephemeris_times'):
+            self.read_image_data()
+        return self._binary_ephemeris_times
+
+
+    @property
+    def binary_lines(self):
+        """
+        Returns the lines of the binary image data. 
+
+        For example, the first entry would be the first line of the image.
+
+        Returns
+        -------
         : list
-          Exposure durations
+          Image lines
         """
-        if not hasattr(self, '_line_scan_rate'):
-            self._line_scan_rate = None
-            lines = []
-            times = []
-            durations = []
-
-            with open(self._file, 'rb') as image_file:
-                bytes_per_record = self.label['RECORD_BYTES']
-                num_records = self.label['FILE_RECORDS']
-                img_start_record = self.label['^IMAGE']
-                img_start_byte = bytes_per_record * (img_start_record - 1) # Offset by one for zero-based records
-                num_img_records = num_records - img_start_record
-                image_file.seek(img_start_byte)
-
-                for record in range(num_img_records):
-                    record_bytes = image_file.read(bytes_per_record)
-                    eph_time = struct.unpack('<d', record_bytes[:8])[0]
-                    exp_dur = struct.unpack('<f', record_bytes[8:12])[0] / 1000
-                    #if record == 0:
-                    # Offset for zero-based corrections, and then offest for ISIS pixel definition
-                    lines.append(record+1-0.5)
-                    times.append(eph_time)
-                    durations.append(exp_dur)
-                # Only add records if exposure duration has changed since the line before
+        if not hasattr(self, '_binary_lines'):
+            self.read_image_data()
+        return self._binary_lines   
+    
+
+    def read_image_data(self):
+        """
+        Reads data off of image and stores in binary_exposure_durations, binary_lines,
+        and binary_ephemeris_times.
+
+        For HRSC, the exposure durations and ephemeris times are imbedded in the binary 
+        data of the image itself. Each line is stored in what is referred to as a 
+        "record" within the image. The label will have the size of each record, 
+        the number of records, and the number of records in the label, so the 
+        beginning of binary data can be calculated.
+
+        For each line/record of the binary data, the first 8 bytes make up the
+        double presicion value of the ephemeris time, with the next 4 bytes
+        making up the float value of the line exposure duration for the
+        associated line. NOTE: The image label specifies MSB_INTEGER as the byte
```

Comment:
Are the images always MSB? Do we need to check the label for this?

Reply:
Looking into this right now. After looking at hrsc2isis, the prefix data is always LSB regardless of the rest of the file. We might need to change this code.

Reply:
The code was treating it as LSB in the driver, but the comment was wrong. Should be all straightened out now.